packages/contracts/test/token/erc20/governance-wrapped-erc20.ts

```diff
@@ -0,0 +1,335 @@
+        ...defaultGovernanceWrappedERC20InitData
+      );
+
+      await expect(
+        governanceToken.initialize(...defaultGovernanceWrappedERC20InitData)
+      ).to.be.revertedWith(OZ_ERRORS.ALREADY_INITIALIZED);
+    });
+
+    it('sets the wrapped token name and symbol', async () => {
+      governanceToken = await GovernanceWrappedERC20.deploy(
+        ...defaultGovernanceWrappedERC20InitData
+      );
+
+      expect(await governanceToken.name()).to.eq(governanceWrappedERC20Name);
+      expect(await governanceToken.symbol()).to.eq(
+        governanceWrappedERC20Symbol
+      );
+    });
+  });
+
+  describe('supportsInterface:', async () => {
+    it('it supports all inherited interfaces', async () => {
+      governanceToken = await GovernanceWrappedERC20.deploy(
+        ...defaultGovernanceWrappedERC20InitData
+      );
+
+      await Promise.all(
+        [
+          'IGovernanceWrappedERC20',
+          'IERC165Upgradeable',
+          'IERC20Upgradeable',
+          'IERC20PermitUpgradeable',
+          'IVotesUpgradeable',
+        ].map(async interfaceName => {
+          const iface = new ethers.utils.Interface(
+            // @ts-ignore
+            (await hre.artifacts.readArtifact(interfaceName)).abi
+          );
+          expect(await governanceToken.supportsInterface(getInterfaceID(iface)))
+            .to.be.true;
+        })
+      );
+      // We must check `IERC20MetadataUpgradeable` separately as it inherits from `IERC20Upgradeable` and we cannot get the isolated ABI.
+      const ierc20MetadataInterface = new ethers.utils.Interface([
+        'function name()',
+        'function symbol()',
+        'function decimals()',
+      ]);
+      expect(
+        await governanceToken.supportsInterface(
+          getInterfaceID(ierc20MetadataInterface)
+        )
+      ).to.be.true;
+    });
+  });
+
+  describe('depositFor', async () => {
+    beforeEach(async function () {
+      governanceToken = await GovernanceWrappedERC20.deploy(
+        ...defaultGovernanceWrappedERC20InitData
+      );
+    });
+
+    it('reverts if the amount is not approved', async () => {
+      const erc20Balance = await erc20.balanceOf(signers[0].address);
+      await expect(
+        governanceToken.depositFor(signers[0].address, erc20Balance)
+      ).to.be.revertedWith(OZ_ERRORS.ERC20_INSUFFICIENT_ALLOWANCE);
+    });
+
+    it('deposits an amount of tokens', async () => {
+      // check old balances
+      const erc20Balance = await erc20.balanceOf(signers[0].address);
+      const governanceTokenBalance = await governanceToken.balanceOf(
+        signers[0].address
+      );
+      expect(erc20Balance).to.eq(123);
+      expect(governanceTokenBalance).to.eq(0);
+
+      // wrap 100 erc20 tokens from signers[0] for signers[0]
+      const depositAmount = 100;
+      await erc20.approve(governanceToken.address, depositAmount);
+      await governanceToken.depositFor(signers[0].address, depositAmount);
+
```

Comment:
I don't know where to include it, but `depositFor` inside calls `_mint` which in the end, calls `_mint`. I'd add the same test that I explained in the `GovernanceERC20` test file.

Reply:
Agreed with @novaknole to be done in another task